src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -426,85 +426,113 @@ private async Task<Solution> AddSingleSymbolInitializationAsync(
-            // is appropriate placed before/after an existing field/property.  We'll try
-            // to preserve the same order for fields/properties that we have for the constructor
-            // parameters.
-            editor.ReplaceNode(
-                typeDeclaration,
-                (currentTypeDecl, _) =>
-                {
-                    if (fieldOrProperty is IPropertySymbol property)
-                    {
-                        return codeGenerator.AddProperty(
-                            currentTypeDecl, property,
-                            codeGenerator.GetInfo(GetAddContext<IPropertySymbol>(parameter, blockStatement, typeDeclaration, cancellationToken), options, root.SyntaxTree.Options),
-                            cancellationToken);
-                    }
-                    else if (fieldOrProperty is IFieldSymbol field)
-                    {
-                        return codeGenerator.AddField(
-                            currentTypeDecl, field,
-                            codeGenerator.GetInfo(GetAddContext<IFieldSymbol>(parameter, blockStatement, typeDeclaration, cancellationToken), options, root.SyntaxTree.Options),
-                            cancellationToken);
-                    }
-                    else
-                    {
-                        throw ExceptionUtilities.Unreachable();
-                    }
-                });
-        }
+        // First, add the field/property to the solution if they're a newly generated member and not a pre-existing one.
+        var (documentWithMemberAdded, currentParameter, currentFieldOrProperty) = await AddMissingFieldOrPropertyAsync(
+            document, constructorDeclaration, blockStatement, parameter, fieldOrProperty, cancellationToken).ConfigureAwait(false);
 
         var initializeParameterService = document.GetRequiredLanguageService<IInitializeParameterService>();
-        initializeParameterService.AddAssignment(
-            constructorDeclaration, blockStatement, parameter, fieldOrProperty, editor);
 
-        // If the user had a property that has 'throw NotImplementedException' in it, then remove those throws.
-        var currentSolution = document.Project.Solution;
-        if (isThrowNotImplementedProperty)
+        var solutionWithAssignmentAdded = documentWithMemberAdded.Project.Solution;
+        if (currentParameter != null && currentFieldOrProperty != null)
         {
-            var declarationService = document.GetRequiredLanguageService<ISymbolDeclarationService>();
-            var propertySyntax = await declarationService.GetDeclarations(fieldOrProperty)[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
-            var withoutThrowNotImplemented = RemoveThrowNotImplemented(propertySyntax);
+            // Now, attempt to assign the parameter to that field/property.
+            solutionWithAssignmentAdded = await initializeParameterService.AddAssignmentAsync(
```

Comment:
second step is to add the assignment statement from the parameter to the field (existing or new).